src/Sema/comptime_ptr_access.zig

```diff
@@ -855,8 +855,12 @@ fn prepareComptimePtrStore(
             => break, // terminal types (no sub-values)
             .optional => break, // this can only be a pointer-like optional so is terminal
             .array => {
+                const len = ip.indexToKey(cur_ty.toIntern()).array_type.len;
                 const elem_ty = cur_ty.childType(zcu);
                 const elem_size = try elem_ty.abiSizeSema(pt);
+                if (len == 0 or elem_size == 0) {
+                    break; // terminal array (0-sized array)
```

Comment:
What do you mean by "terminal array"? Also this should likely use `lenIncludingSentinel()`.